src/devices/UFireIse/UFireIse.cs

```diff
@@ -0,0 +1,415 @@
+        /// Initializes a new instance of the <see cref="UFireIse"/> class.
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device to be used</param>
+        public UFireIse(I2cDevice i2cDevice)
+        {
+            _device = i2cDevice;
+        }
+
+        /// <summary>
+        /// Read a value from the ISE Probe Interface, typical measure are in the millivolt range.
+        /// </summary>
+        /// <returns>value from ISE Probe Interface, typical measure are in the millivolt range. On error it return -1 as value</returns>
+        public ElectricPotential Read()
+        {
+            SendCommand((byte)Command.ISE_MEASURE_MV);
+
+            DelayHelper.DelayMilliseconds(IseMvMeasureTime, allowThreadYield: true);
+
+            _measurement = ReadFloat(Register.ISE_MV_REGISTER);
+            _measurement = Convert.ToSingle(Math.Round(_measurement, 2));
+
+            return new ElectricPotential(_measurement, UnitsNet.Units.ElectricPotentialUnit.Millivolt);
+        }
+
+        /// <summary>
+        /// Read temperature
+        /// </summary>
+        /// <returns>Temperature. Temperature range: -2 to 35 C. Temperature Precision: 0.05 C</returns>
+        public Temperature ReadTemperature()
+        {
+            SendCommand((byte)Command.ISE_MEASURE_TEMP);
+
+            DelayHelper.DelayMilliseconds(IseTemperatureMeasureTime, allowThreadYield: true);
+            _temperature = new Temperature(ReadFloat(Register.ISE_TEMP_REGISTER), UnitsNet.Units.TemperatureUnit.DegreeCelsius);
+
+            return _temperature;
+        }
+
+        /// <summary>
+        /// It sets the a temperature used as compensation. The compensation temperature is to correlate the mV input to the correct pH value, see https://assets.tequipment.net/assets/1/26/Yokogawa_Temperature_Compensation.pdf
+        /// </summary>
+        /// <param name="temp">The temperature used for compensation</param>
+        public void SetTemperatureCompensation(Temperature temp)
+        {
+            WriteRegister(Register.ISE_TEMP_REGISTER, Convert.ToSingle(temp.DegreesCelsius));
+            _temperature = temp;
+        }
+
+        /// <summary>
+        /// Calibrates the probe using a single solution. Put the probe in a solution where the pH (Power of Hydrogen) is know.
+        /// </summary>
+        /// <param name="solution">The known pH value in mV. Range: -1024 mV to 1024 mV</param>
+        public void CalibrateFromSingleValue(ElectricPotential solution)
+        {
+            if (solution.Millivolts >= -1024 && solution.Millivolts <= 1024)
+            {
+                WriteRegister(Register.ISE_SOLUTION_REGISTER, Convert.ToSingle(solution.Millivolts));
+                SendCommand((byte)Command.ISE_CALIBRATE_SINGLE);
+
+                DelayHelper.DelayMilliseconds(IseMvMeasureTime, allowThreadYield: true);
+            }
+
+            throw new ArgumentOutOfRangeException("The value range is -1024 mV to 1024 mV");
```

Comment:
this will now always throw (missing `else` - make sure to use braces around throw)